### PR TITLE
Added SHIR Deployment via Custom Data

### PR DESCRIPTION
**Pull request by marvinbuss**
**This PR fixes**
* Install SHIR via Custom Data and Custom Script Extension
* Removed unnecessary input parameters in parameter files, portal UI & docs
* Simplified CI/CD pipelines
* Removed old outputs
* Updated VMSS to Windows Server 2022
* Removed Artifact Storage

### CODE REVIEW

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Comment by marvinbuss** [comment]
[Preview Link](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fdata-landing-zone%2Fmarvinbuss%2Fshir_custom_data%2Finfra%2Fmain.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fdata-landing-zone%2Fmarvinbuss%2Fshir_custom_data%2Fdocs%2Freference%2Fportal.dataLandingZone.json)

File: .github/linters/.arm-ttk.psd1 [hunk 1]
```diff
@@ -9,5 +9,6 @@
         'DependsOn Best Practices',
         'Outputs Must Not Contain Secrets',
         'IDs Should Be Derived From ResourceIDs'
```
Comment (amanjeetsingh) [inline .github/linters/.arm-ttk.psd1:11]: should there be a comma at the end? if not, pls ignore / close this comment. 
thanks

File: docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
 | `dataIntegration002SubnetAddressPrefix` | Specifies the address space of the subnet that is used for Data Integration 002. | `10.1.7.0/24` |
 | `dataProduct001SubnetAddressPrefix` | Specifies the address space of the subnet that is used for Data Product 001. | `10.1.8.0/24` |
 | `dataProduct002SubnetAddressPrefix` | Specifies the address space of the subnet that is used for Data Product 002. | `10.1.9.0/24` |
-| `dataManagementZoneVnetId` | Specifies the resource Id of the vnet in the Data Management Zone. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Network/virtualNetworks/{vnet-name}` |
+| `dataManagementZoneVnetId` | Specifies the resource ID of the vnet in the Data Management Zone. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Network/virtualNetworks/{vnet-name}` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md:78]: 'resource ID' change to 'Resource ID' 

vnet to VNet  to keep it consistent

Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md:78]: resource ID to Resource ID

vnet to VNet

**Review by amanjeetsingh — COMMENTED** [review]
Minor tweaks required; see comments. Thanks

File: .github/linters/.arm-ttk.psd1 (hunk 1 above)
Comment (marvinbuss) [reply .github/linters/.arm-ttk.psd1:11]: You are right, it is optional, but I would like to set a comma for consistency reasons in other locations.

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Comment by sasever** [comment]
@marvinbuss the tests are done with:
 parameters: /home/runner/work/data-landing-zone/data-landing-zone/infra/params.dev.json administratorPassword="***" deploySelfHostedIntegrationRuntimes=false

Are you sure that whole custom data deployment workflow is executed in test?

**Comment by marvinbuss** [comment]
@sasever please check under "Files changed": 
image
This was updated in this branch.

Full test deployment with ADO was also executed:
image

**Review by marvinbuss — COMMENTED** [review]
LGTM!
Thanks @sasever for the heads up and for noticing these small bugs!

**Review by sasever — APPROVED** [review]
Legitimate. Thank you for fixing the last changes.

**Pull request merged**